app/assets/javascripts/components/components/status_content.jsx

```diff
@@ -93,7 +93,7 @@ const StatusContent = React.createClass({
     const { status } = this.props;
     const { hidden } = this.state;
 
-    const content = { __html: emojify(status.get('content')) };
+    const content = { __html: emojify(status.get('content').replace(/\s*(<p>|<\/p>)\s*/g, '$1').replace(/\s{2}/g, ' &nbsp;')) };
```

Comment:
Why not double `&nbsp;`?

Doing all of this in this JS piece, what happens to the public pages that don't use this code?

You may use [this](https://mstdn.jp/@Maxillaria/414078) as a landmark of what should not be broken.